module/synchronization/core.go

```diff
@@ -206,15 +213,17 @@ func (c *Core) queueByHeight(height uint64) {
 
 // queueByBlockID queues a request for a block by block ID, only if no
 // equivalent request has been queued before.
-func (c *Core) queueByBlockID(blockID flow.Identifier) {
+func (c *Core) queueByBlockID(blockID flow.Identifier, height uint64) {
```

Comment:
with `localFinalizedHeight`, we will only keep requesting blocks that are above the finalized height. So we could also drop blocks that is below or equal to `localFinalizedHeight`.